### ISSUE TITLE
Inconsistent ConfigParameters Behavior: Missing samp_freq Field

### ISSUE THREAD

**Issue by SeismoFelix** [noisepy/NoisePy#329]
Hi, sorry to bother you again with this issue.

I recently tried to help someone install NoisePy, but the same versioning issue arose again. When running the [main tutorial](https://seisscoped.org/HPS-book/chapters/noise/noisepy_monitoring_demo.html), I encountered the following error:

> ValueError: "ConfigParameters" object has no field "samp_freq"

This error occurs when executing the cell for setting configuration parameters for monitoring:

> config = ConfigParameters()

It seems the installed version of the code does not include the samp_freq parameter. This is strange because I have another installation where this error does not occur, and the tutorial works as expected.

Investigation So Far:
1. I inspected [monitoring_utils.py](https://github.com/noisepy/NoisePy/blob/dd835cad251e22cb558ddd742c964190c4c749dc/src/noisepy/monitoring/monitoring_utils.py#L24), but samp_freq is not explicitly declared in the ConfigParameters_monitoring class.

2. I also checked /noisepy/seis/io/datatypes.py in the local installation, but I couldn't locate samp_freq there either. Additionally, I noticed that the io directory under [seis](https://github.com/noisepy/NoisePy/tree/dd835cad251e22cb558ddd742c964190c4c749dc/src/noisepy/seis) does not appear to exist in the GitHub repository, making it difficult to compare the installed files with the repository version.

Could you please help clarify:

1. Where the samp_freq parameter for the ConfigParameters class is declared or inherited from?
2. Why this inconsistency might occur between installations when using the same NoisePy tutorial?

Any guidance would be greatly appreciated. Thank you so much for your help, and I apologize for the inconvenience!

**Comment by niyiyu** [comment]
Hi @SeismoFelix,

Sorry for the issue you are facing. We are still going through some major change to the package. It's mostly done, but there might still be some changes to the parameters or variables. In short, the problem you are facing comes from a [recent change](https://github.com/noisepy/noisepy-io/pull/34) on IO that replace `samp_freq` into `sampling_rate` to match the obspy convention.

> I recently tried to help someone install NoisePy, but the same versioning issue arose again. When running the [main tutorial](https://seisscoped.org/HPS-book/chapters/noise/noisepy_monitoring_demo.html), I encountered the following error:
> 
> > ValueError: "ConfigParameters" object has no field "samp_freq"
> 
> This error occurs when executing the cell for setting configuration parameters for monitoring:
> 
> > config = ConfigParameters()
> 
> It seems the installed version of the code does not include the samp_freq parameter. This is strange because I have another installation where this error does not occur, and the tutorial works as expected.

We apologize that the tutorial on HPS-book is not updated since the change. It was trying to set the parameter `samp_freq`, which does not exist anymore. We have updated the notebook on HPS-book. We also recommend using [this](https://noisepy.github.io/NoisePy/README.html) documentation, which is always synced with our packages. 

> 1. I inspected [monitoring_utils.py](https://github.com/noisepy/NoisePy/blob/dd835cad251e22cb558ddd742c964190c4c749dc/src/noisepy/monitoring/monitoring_utils.py#L24), but samp_freq is not explicitly declared in the ConfigParameters_monitoring class.

The `ConfigParameters_monitoring` is specifically for monitoring. It makes sense there isn't either `samp_freq` or `sampling_rate` field.

> 2. I also checked /noisepy/seis/io/datatypes.py in the local installation, but I couldn't locate samp_freq there either. Additionally, I noticed that the io directory under [seis](https://github.com/noisepy/NoisePy/tree/dd835cad251e22cb558ddd742c964190c4c749dc/src/noisepy/seis) does not appear to exist in the GitHub repository, making it difficult to compare the installed files with the repository version.

We put IO as a separate repository for easier maintenance. You can find the repo [here](https://github.com/noisepy/noisepy-io), and the place `ConfigParameters` is defined [here](https://github.com/noisepy/noisepy-io/blob/f7805eb933f591ed10753a4b31763560b9691da9/src/noisepy/seis/io/datatypes.py#L146).

Hopefully this helps solve your problem.

**Comment by SeismoFelix** [comment]
Thanks a lot @niyiyu 

I was able to run the tutorial with the newest version of noisepy. I just made a minor modifications to the paremeters that still belonged to the deprecated version: 

> config.net_list = ["CI"] -> config.networks = ["CI"]

I also updated some lines that still call the old parameters:

> raw_store = SCEDCS3DataStore(S3_DATA, catalog, channel_filter(**config.net_lis**t, config.stations, ["BH?", "HH?"]),timerange, storage_options=S3_STORAGE_OPTIONS) # Store for reading raw data from S3 bucket

->

> raw_store = SCEDCS3DataStore(S3_DATA, catalog, channel_filter(**config.networks**, config.stations, ["BH?", "HH?"]), timerange, storage_options=S3_STORAGE_OPTIONS)

and also in defining the dt before calculating the reference curve

> dt=1/config.samp_freq -> dt=1/config.sampling_rate

Now I am trying to conciliate the legacy parameters with the newest ones for controling the number of dv/v measurements. 
In this new version, as is, I am getting 4044 measurements for a month of data:

<img width="1235" alt="Screenshot 2024-11-23 at 8 21 35 PM" src="https://github.com/user-attachments/assets/fd1535e0-209b-47ac-9157-ebec45729cf2">

Which seems to be excesive compared to the previous version of the tutorial, where there were 40 samples:

<img width="876" alt="Screenshot 2024-11-23 at 8 22 21 PM" src="https://github.com/user-attachments/assets/8f621f3f-261a-4971-aab9-4353a39286ac">

I changed config.substack_windows = 2 (I assume is the new parameter in exchange of config.substack_len) and I got 30 windows, but the reference curve output looks significatly different from the old version I run a couple of weeks ago. I am hesitant that 10 samples changed the outcome too much. 

**New version:**
<img width="1100" alt="Screenshot 2024-11-23 at 8 24 57 PM" src="https://github.com/user-attachments/assets/3c0c5229-5c4e-4897-b8d3-185305da9f2d">

**Old version:**
<img width="775" alt="Screenshot 2024-11-23 at 8 25 09 PM" src="https://github.com/user-attachments/assets/ea263c5e-c60c-4fd0-966c-335961e72f3d">

After the modifications I suggest, with the today noisepy version, the tutorial is functional. Although the outcome is different from what I got some weeks ago. But I believe this is a matter of being familiar with the new parameters. So, feel free to close this issue if you consider  it convenient. 

I appreciate a lot how responsive have you been. I am still climbing the learning curve and I will keep this issue update it with the aim to help my colleagues that also are learning.

Cheers

**Comment by SeismoFelix** [comment]
Good news (to me :) )

I was able to to run the tutorial monitoring notebook getting the same results as the obtained ones in the previous version of noisepy. It took me a while to realize than `config.substack_len` is defined internally now as 
`config.substack_len = self.substack_windows * self.cc_len`. So I had to define` config.substack_windows = 24` and `cc_len = 600 ` as it was. Then I could get the 6  segments/windows as in the previous version and reproducing the same curves. 

Since you are still modifying certain parameters, let's see how this end up. But I will try to move forward with this version. 

Thanks a lot for your help!
